slatedb/src/compactor.rs

```diff
@@ -394,6 +394,94 @@ impl CompactorEventHandler {
     /// Emits the current compaction state and per-job progress.
     fn handle_log_ticker(&self) {
         self.log_compaction_state();
+        self.log_compaction_throughput();
+    }
+
+    /// Logs per-job compaction progress and updates aggregate throughput metrics.
+    fn log_compaction_throughput(&self) {
+        let current_time = self.system_clock.now();
+        let current_time_ms = current_time.timestamp_millis() as u64;
+        let db_state = self.state.db_state();
+        let mut total_bytes = 0u64;
+        let mut total_throughput = 0.0;
+
+        for compaction in self.state.compactions() {
+            let estimated_source_bytes =
+                Self::calculate_estimated_source_bytes(compaction, db_state);
+            total_bytes += estimated_source_bytes;
+
+            // Calculate throughput using ULID timestamp
+            let start_time_ms = compaction
+                .id()
+                .datetime()
+                .duration_since(std::time::UNIX_EPOCH)
+                .expect("invalid duration")
+                .as_millis() as u64;
+            let elapsed_secs = if start_time_ms > 0 {
+                (current_time_ms as f64 - start_time_ms as f64) / 1000.0
+            } else {
+                0.0
+            };
+            let throughput = if elapsed_secs > 0.0 {
+                compaction.bytes_processed() as f64 / elapsed_secs
+            } else {
+                0.0
+            };
+            total_throughput += throughput;
```

Comment:
There's a bit of a stats danger in here. We're summing averages with unlike denominators (different elapse_secs for each compaction). This can be misleading. One could argue that we should instead sum all bytes and sum all elapsed_secs and _then_ do the division once. I think that's a slightly more accurate (depending on how you define accurate here) approach.